virtualcluster/pkg/syncer/resources/pod/mutatorplugin/podkubeapiaccessmutator_test.go

```diff
@@ -0,0 +1,204 @@
+limitations under the License.
+*/
+
+package mutatorplugin
+
+import (
+	"strings"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"sigs.k8s.io/cluster-api-provider-nested/virtualcluster/pkg/syncer/constants"
+	"sigs.k8s.io/cluster-api-provider-nested/virtualcluster/pkg/syncer/conversion"
+	"sigs.k8s.io/cluster-api-provider-nested/virtualcluster/pkg/syncer/util/featuregate"
+	util "sigs.k8s.io/cluster-api-provider-nested/virtualcluster/pkg/syncer/util/test"
+	uplugin "sigs.k8s.io/cluster-api-provider-nested/virtualcluster/pkg/util/plugin"
+)
+
+func tenantKubeAPIAccessPod(name, namespace, uid string) *corev1.Pod {
+	p := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			UID:       types.UID(uid),
+		},
+		Spec: corev1.PodSpec{
+			ServiceAccountName: "default",
+			InitContainers: []corev1.Container{
+				{
+					Image: "busybox",
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "kube-api-access-l945t",
+							MountPath: "/var/run/secrets/kubernetes.io/serviceaccount",
+							ReadOnly:  true,
+						},
+					},
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Image: "busybox",
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "kube-api-access-l945t",
+							MountPath: "/var/run/secrets/kubernetes.io/serviceaccount",
+							ReadOnly:  true,
+						},
+					},
+				},
+			},
+			Volumes: []corev1.Volume{
+				{
+					Name: "kube-api-access-l945t",
+					VolumeSource: corev1.VolumeSource{
+						Projected: &corev1.ProjectedVolumeSource{
+							Sources: []corev1.VolumeProjection{
+								{
+									ServiceAccountToken: &corev1.ServiceAccountTokenProjection{
+										Path: "token",
+									},
+									ConfigMap: &corev1.ConfigMapProjection{
+										Items: []corev1.KeyToPath{
+											{
+												Key:  "ca.crt",
+												Path: "ca.crt",
+											},
+										},
+										LocalObjectReference: corev1.LocalObjectReference{Name: constants.RootCACertConfigMapName},
+									},
+									DownwardAPI: &corev1.DownwardAPIProjection{
+										Items: []corev1.DownwardAPIVolumeFile{
+											{
+												Path: "namespace",
+												FieldRef: &corev1.ObjectFieldSelector{
+													APIVersion: "v1",
+													FieldPath:  "metadata.namespace",
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	return p
+}
+
+func TestPodKubeAPIAccessMutatorPlugin_Mutator(t *testing.T) {
+	defer util.SetFeatureGateDuringTest(t, featuregate.DefaultFeatureGate, featuregate.KubeAPIAccessSupport, true)()
+
+	tests := []struct {
+		name                  string
+		pPod                  *corev1.Pod
+		existingObjectInSuper []runtime.Object
```

Comment:
It'd be easier if you add an expected Pod after mutation as a field in this struct.

Reply:
hi, @ravisantoshgudimetla. I thought about it. This is not easy to implement, because the names kube-api-access-xxxx here are randomly generated by `names.SimpleNameGenerator.GenerateName`.

Reply:
I see. No issues then. I was actually referring to the contents not the name.